qiskit/quantum_info/operators/symplectic/clifford.py

```diff
@@ -145,73 +157,143 @@ def name(self):
+        "and will be removed no sooner than 3 months after the release date. "
+        "Use Clifford.paulis method instead.",  # pylint:disable=bad-docstring-quotes
+    )
     def table(self, value):
         """Set the stabilizer table"""
         # Note this setter cannot change the size of the Clifford
         # It can only replace the contents of the StabilizerTable with
         # another StabilizerTable of the same size.
         if not isinstance(value, StabilizerTable):
             value = StabilizerTable(value)
-        self._table._array[:, :] = value._table._array
-        self._table._phase[:] = value._table._phase
+        self.paulis = value
 
     @property
+    @deprecate_function(
+        "The Clifford.stabilizer method is deprecated as of Qiskit Terra 0.19.0 "
+        "and will be removed no sooner than 3 months after the release date. "
+        "Use Clifford.stabilizers method instead.",  # pylint:disable=bad-docstring-quotes
+    )
     def stabilizer(self):
         """Return the stabilizer block of the StabilizerTable."""
-        return StabilizerTable(self._table[self.num_qubits : 2 * self.num_qubits])
+        return StabilizerTable(self.table[self.num_qubits : 2 * self.num_qubits])
 
     @stabilizer.setter
+    @deprecate_function(
+        "The Clifford.stabilizer method is deprecated as of Qiskit Terra 0.19.0 "
+        "and will be removed no sooner than 3 months after the release date. "
+        "Use Clifford.paulis property instead.",  # pylint:disable=bad-docstring-quotes
+    )
     def stabilizer(self, value):
         """Set the value of stabilizer block of the StabilizerTable"""
         inds = slice(self.num_qubits, 2 * self.num_qubits)
-        self._table.__setitem__(inds, value)
+        self._paulis.__setitem__(inds, value)
 
     @property
+    @deprecate_function(
+        "The Clifford.destabilizer method is deprecated as of Qiskit Terra 0.19.0 "
+        "and will be removed no sooner than 3 months after the release date. "
+        "Use Clifford.destabilizers method instead.",  # pylint:disable=bad-docstring-quotes
+    )
     def destabilizer(self):
         """Return the destabilizer block of the StabilizerTable."""
-        return StabilizerTable(self._table[0 : self.num_qubits])
+        return StabilizerTable(self.table[0 : self.num_qubits])
 
     @destabilizer.setter
+    @deprecate_function(
+        "The Clifford.destabilizer method is deprecated as of Qiskit Terra 0.19.0 "
+        "and will be removed no sooner than 3 months after the release date. "
+        "Use Clifford.paulis property instead.",  # pylint:disable=bad-docstring-quotes
+    )
     def destabilizer(self, value):
         """Set the value of destabilizer block of the StabilizerTable"""
         inds = slice(0, self.num_qubits)
-        self._table.__setitem__(inds, value)
+        self._paulis.__setitem__(inds, value)
+
+    def stabilizers(self):
```

Comment:
So we can get the stabilizers and destabilizers but can't set them?
I think that in some cases also setting them can be useful.

Reply:
If I remember correctly, there was a previous discussion about this or something closely related in an earlier review.  I'm not sure if there's a case where you'd need to set the stabilizers or destabilizers separately to each other and can't just create a new `Clifford` object - that's certainly safer.

At any rate, part of the problem here is that with the two parts no longer stored as two separate components, we have to construct new objects when you ask for `stabilizers` or `destabilizers`, so it's probably better as a performance hint to the programmer that we use a proper method for getting the stabilizers, to indicate the cost of repeating it.